File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stk=[]
        for i in tokens:
            if i in ("+", "-", "*", "/"):
                b=stk.pop()
                a=stk.pop()
                if i=="+":
                    stk.append(a+b)
                if i=="-":
                    stk.append(a-b)
                if i=="*":
                    stk.append(a*b)
                if i=="/":
                    c=a/b
                    if c>=0:
                        stk.append(math.floor(c))
                    else:
                        stk.append(math.ceil(c))
            else:
                stk.append(int(i))
        return stk[-1]

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stk = new Stack<>();
        for(String c: tokens){
            if(c.equals("+")) stk.push(stk.pop()+stk.pop());
            else if(c.equals("-")) stk.push((stk.pop()*-1)+stk.pop());
            else if(c.equals("*")) stk.push(stk.pop()*stk.pop());
            else if(c.equals("/")){
                int a = stk.pop(), b = stk.pop();
                stk.push(b/a);
            }
            else stk.push(Integer.parseInt(c));
        }
        return stk.peek();
    }
}




// class Solution {
//     public int evalRPN(String[] tokens) {
//         Stack<String> stk = new Stack<>();
//         for(var t: tokens){
//             if(t.equals("+")){
//                 int sec = Integer.parseInt(stk.pop());
//                 int first = Integer.parseInt(stk.pop());
//                 stk.push(Integer.toString(first+sec));
//             }
//             else if(t.equals("-")){
//                 int sec = Integer.parseInt(stk.pop());
//                 int first = Integer.parseInt(stk.pop());
//                 stk.push(Integer.toString(first-sec));
//             }
//             else if(t.equals("*")){
//                 int sec = Integer.parseInt(stk.pop());
//                 int first = Integer.parseInt(stk.pop());
//                 stk.push(Integer.toString(first*sec)); 
//             }
//             else if(t.equals("/")){
//                 int sec = Integer.parseInt(stk.pop());
//                 int first = Integer.parseInt(stk.pop());
//                 stk.push(Integer.toString(first/sec));                
//             }
//             else{
//                 stk.push(t);
//             }
//         }
//         return Integer.parseInt(stk.pop());
//     }
// }